configure.ac

```diff
@@ -2621,12 +2621,22 @@ fi
 LIB_SOCKET_NSL
 AX_HARDEN_CC_COMPILER_FLAGS
 
-# link to ws2_32 if on mingw
+# if mingw then link to ws2_32 for sockets
 case $host_os in
-    *mingw32)
+    mingw*)
         LDFLAGS="$LDFLAGS -lws2_32" ;;
 esac
 
+# add wolfSSL defines for shared/static
+if test "$enable_shared" = "yes"
+then
+    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_SHARED"
+fi
+if test "$enable_static" = "yes"
+then
+    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_STATIC"
+fi
+
```

Comment:
@toddouska

Currently the ./configure allows both shared and static to be enabled. Is that okay? Seems to me like they are mutually exclusive (only one should be enabled). 

Thinking this should be:
```
# add wolfSSL defines for shared or static
if test "$enable_shared" = "yes"
then
    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_SHARED"
else
    if test "$enable_static" = "yes"
    then
        AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_STATIC"
    fi
fi
```

And maybe we should add a trap against that?